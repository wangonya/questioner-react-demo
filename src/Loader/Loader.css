.loading {
  visibility: visible;
  -webkit-animation: fadein 0.5s;
  animation: fadein 0.5s;
}

.spinner {
  width: 80px;
  height: 80px;
  border: 2px solid #f3f3f3;
  border-top: 3px solid #f0776c;
  border-radius: 100%;
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  margin: auto;
  animation: spin 1s infinite linear;
  z-index: 999;
}

.loading:before {
  content: "";
  display: block;
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: rgba(0, 0, 0, 0.1);
  z-index: 998;
}

@keyframes spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
