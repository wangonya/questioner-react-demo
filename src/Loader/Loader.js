import React from "react";

import "./Loader.css";

const Loader = () => {
  return (
    <div className="loading">
      <span className="spinner" />
    </div>
  );
};

export default Loader;
