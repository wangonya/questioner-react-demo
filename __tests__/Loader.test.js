import React from "react";
import { shallow } from "enzyme";

import Loader from ".././src/Loader/Loader";

describe("<Loader /> component", () => {
  const wrapper = shallow(<Loader />);

  test("renders without crashing", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
