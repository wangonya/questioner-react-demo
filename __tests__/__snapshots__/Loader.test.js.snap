// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> component renders without crashing 1`] = `ShallowWrapper {}`;
